From 327fd5ea57086596ceff247326882b8935c5d42e Mon Sep 17 00:00:00 2001
From: Omri Itach <omrii@marvell.com>
Date: Tue, 3 May 2016 11:15:30 +0300
Subject: [PATCH 1121/1240] dts: 8040: update cp110-slave hierarchy

cp110-slave.dtsi:
- Align with cp110-master.dtsi
- disable S@R from 2nd CP - not needed
- disabled utmi,utmi1,mdio (mistakenly enabled in cp110.dtsi)

A8040:
include cp110-slave.dtsi:
- enable comphy lane2: 10G (KR)
- rest of interfaces currently disabled

Change-Id: I2a083d6aa145f3c0384c1b8e570af64acc4f1dfe
Signed-off-by: Omri Itach <omrii@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/29451
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Haim Boot <hayim@marvell.com>
---
 arch/arm/dts/armada-80x0-db.dtsi | 137 +++++++++++++++++++++++++++++++++++++++
 arch/arm/dts/armada-80x0.dtsi    |   2 +-
 arch/arm/dts/cp110-slave.dtsi    |  31 ++++-----
 arch/arm/dts/cp110.dtsi          |   4 +-
 4 files changed, 154 insertions(+), 20 deletions(-)

diff --git a/arch/arm/dts/armada-80x0-db.dtsi b/arch/arm/dts/armada-80x0-db.dtsi
index 9bb6bca..a3983fd 100644
--- a/arch/arm/dts/armada-80x0-db.dtsi
+++ b/arch/arm/dts/armada-80x0-db.dtsi
@@ -190,7 +190,144 @@
 	};
 
 	cp110-slave {
+		internal-regs {
+			spi0@700600 {
+				status = "disabled";
+			};
+			spi1@700680 {
+				status = "disabled";
+			};
+			i2c@701000 {
+				status = "disabled";
+			};
+			i2c@211000 { /* MSS_i2c0 */
+				status = "disabled";
+			};
+			i2c@211100 { /* MSS_i2c1 */
+				status = "disabled";
+			};
+			usb3@500000 {
+				status = "disabled";
+			};
+			usb3@510000 {
+				status = "disabled";
+			};
+			sata@540000 {
+				status = "disabled";
+			};
+			mmc@780000 {
+				status = "disabled";
+			};
+			pcie-controller {
+				pcie@0,0 {
+					status = "disabled";
+				};
+				pcie@1,0 {
+					status = "disabled";
+				};
+				pcie@2,0 {
+					status = "disabled";
+				};
+			};
+			nand@720000 {
+				status = "disabled";
+			};
+			pinctl@440000 {
+					/* MPP Bus:
+					   [0-31] = 0xff: Keep default CP0_shared_pins:
+						[11] CLKOUT_MPP_11 (out)
+						[23] LINK_RD_IN_CP2CP (in)
+						[25] CLKOUT_MPP_25 (out)
+						[29] AVS_FB_IN_CP2CP (in)
+					   [32,34] SMI
+					   [31]    GPIO: push button/Wake
+					   [35-36] XSMI
+					   [37-38] I2C
+					   [40-41] SATA[0/1]_PRESENT_ACTIVEn
+					   [44-55] RGMII1
+					   [56-62] SD
+					*/
+					/*   0    1    2    3    4    5    6    7    8    9 */
+				pin-func = < 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff
+					     0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff
+					     0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff 0xff
+					     0xff 0    7    0    7    7    7    2    2    0
+					     0    0    0    0    1    1    1    1    1    1
+					     1    1    1    1    1    1    0xE  0xE  0xE  0xE
+					     0xE  0xE  0xE>;
+			};
 
+			comphy {
+				/* Serdes Configuration:
+					Lane 0,1,3,4,5: UNCONNECTED
+					Lane 2: 10G (KR)
+				*/
+				phy0 {
+					phy-type = <PHY_TYPE_UNCONNECTED>;
+					phy-speed = <PHY_SPEED_3_125G>;
+				};
+				phy1 {
+					phy-type = <PHY_TYPE_UNCONNECTED>;
+					phy-speed = <PHY_SPEED_5G>;
+				};
+				phy2 {
+					phy-type = <PHY_TYPE_KR>;
+					phy-speed = <PHY_SPEED_10_3125G>;
+				};
+				phy3 {
+					phy-type = <PHY_TYPE_UNCONNECTED>;
+					phy-speed = <PHY_SPEED_5G>;
+				};
+				phy4 {
+					phy-type = <PHY_TYPE_UNCONNECTED>;
+					phy-speed = <PHY_SPEED_5G>;
+				};
+				phy5 {
+					phy-type = <PHY_TYPE_UNCONNECTED>;
+					phy-speed = <PHY_SPEED_5G>;
+				};
+			};
+			mdio {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				phy4: ethernet-phy@2 {
+					reg = <0>;
+				};
+				phy5: ethernet-phy@3 {
+					reg = <1>;
+				};
+			};
+			gop {
+				mac0 {
+					phy-mode = "sgmii"; /* lane-2 */
+					status = "disabled";
+				};
+				mac2 {
+					phy-mode = "sgmii"; /* lane-0 */
+					phy = <&phy4>;
+					phy-speed = <2500>;
+					status = "disabled";
+				};
+				mac3 {
+					phy-mode = "rgmii"; /* rgmii-1 */
+					phy = <&phy5>;
+					status = "disabled";
+				};
+			};
+			ppv22@000000 {
+				status = "disabled";
+				eth0@010000 {
+					status = "disabled";
+				};
+				eth1@020000 {
+					status = "disabled";
+				};
+				eth2@030000 {
+					status = "disabled";
+				};
+			};
+		};
 	};
 
+
 };
diff --git a/arch/arm/dts/armada-80x0.dtsi b/arch/arm/dts/armada-80x0.dtsi
index 3ec1bee..3b1efb9 100644
--- a/arch/arm/dts/armada-80x0.dtsi
+++ b/arch/arm/dts/armada-80x0.dtsi
@@ -19,7 +19,7 @@
 #include <dt-bindings/comphy/comphy_data.h>
 #include "apn-806-a0.dtsi"
 #include "cp110-master.dtsi"
-/* #include "cp110-slave.dtsi" */
+#include "cp110-slave.dtsi"
 
 / {
 	compatible = "marvell,armada-80x0";
diff --git a/arch/arm/dts/cp110-slave.dtsi b/arch/arm/dts/cp110-slave.dtsi
index f148b80..38fc885 100644
--- a/arch/arm/dts/cp110-slave.dtsi
+++ b/arch/arm/dts/cp110-slave.dtsi
@@ -15,7 +15,7 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  * ***************************************************************************
  */
- /* TODO: align with cp110-master.dtsi */
+
 / {
 	model = "Marvell Armada CPN-110 Slave";
 
@@ -34,25 +34,22 @@
 		};
 	};
 
-	/* Override the above configurations taken from cp110.dtsi */
 	cp110-slave {
 		internal-regs {
-			map-iob {
-				windows = <0x0 0xfb000000 0x0 0x1000000 0x2 /* PEX1_X1 window */
-					0x0 0xfc000000 0x0 0x1000000 0x3 /* PEX2_X1 window */
-					0x0 0xfa000000 0x0 0x1000000 0x4 /* PEX0_X4 window */>;
+			refclk1: refclk {
 			};
-
-			pcie-controller {
-				pcie@0,0 {
-					mem = <0xfa000000 0x1000000>;
-				};
-				pcie@1,0 {
-					mem = <0xfb000000 0x1000000>;
-				};
-				pcie@2,0 {
-					mem = <0xfc000000 0x1000000>;
-				};
+			/* TODO: disable utmi0,utmi1,mdio in cp110.dtsi; enable in board dtsi */
+			utmi0@580000 {
+				status = "disabled";
+			};
+			utmi1@581000 {
+				status = "disabled";
+			};
+			mdio0@12A200 {
+				status = "disabled";
+			};
+			sar-reg {
+				status = "disabled";
 			};
 
 			gop {
diff --git a/arch/arm/dts/cp110.dtsi b/arch/arm/dts/cp110.dtsi
index a9f4bae..4b5e921 100644
--- a/arch/arm/dts/cp110.dtsi
+++ b/arch/arm/dts/cp110.dtsi
@@ -39,7 +39,7 @@ spi0@700600 {
 	clock = <&refclk>;
 	cpol-cpha-cs-bitmap = <0>; /* bit i is set if the CPOL and CPHA
 				      of CS-i is enabled or not */
-	status = "disable";
+	status = "disabled";
 };
 
 spi1@700680 {
@@ -49,7 +49,7 @@ spi1@700680 {
 	clock = <&refclk>;
 	cpol-cpha-cs-bitmap = <0>; /* bit i is set if the CPOL and CPHA
 				      of CS-i is enabled or not */
-	status = "disable";
+	status = "disabled";
 };
 i2c@701000 { /* i2c0 */
 	compatible = "marvell,mvebu-i2c";
-- 
1.9.1

